File: src/cpp/cdm/io/protobuf/PBEnums.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
PUSH_PROTO_WARNINGS()
#include "pulse/cdm/bind/Enums.pb.h"
POP_PROTO_WARNINGS()

const std::string& eSide_Name(eSide m)
{
  return CDM_BIND::eSide_Name((CDM_BIND::eSide)m);
}

const std::string& eGate_Name(eGate m)
{
  return CDM_BIND::eGate_Name((CDM_BIND::eGate)m);
}

const std::string& eSwitch_Name(eSwitch m)
{
  return CDM_BIND::eSwitch_Name((CDM_BIND::eSwitch)m);
}

const std::string& eCharge_Name(eCharge m)
{
  return CDM_BIND::eCharge_Name((CDM_BIND::eCharge)m);
}

File: src/csharp/pulse/cdm/io/PBEnums.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

namespace Pulse.CDM
{
  static class eEnum
  {
    public static string Name(this eSwitch s)
    {
      return ((pulse.cdm.bind.eSwitch)(int)s).ToString();
    }
    public static string Name(this eBreathState w)
    {
      return ((pulse.cdm.bind.eBreathState)(int)w).ToString();
    }

    public static string Name(this eDefaultType t)
    {
      return ((pulse.cdm.bind.eDefaultType)(int)t).ToString();
    }

    public static string Name(this eDriverWaveform w)
    {
      return ((pulse.cdm.bind.eDriverWaveform)(int)w).ToString();
    }

  }
}